packages/preferences/src/browser/UI/settings-ui-plus-editor.tsx

```diff
@@ -0,0 +1,508 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { postConstruct, injectable, inject } from 'inversify';
+import {
+    ReactWidget,
+    PreferenceService,
+    PreferenceSchemaProvider,
+    PreferenceDataProperty,
+    JsonType,
+    PreferenceScope,
+} from '@theia/core/lib/browser';
+import * as React from 'react';
+import { MessageService } from '@theia/core';
+import { PreferenceConfigurations } from '@theia/core/lib/browser/preferences/preference-configurations';
+import { debounce, isEmpty, pickBy } from 'lodash';
+import { fuzzyFilter } from './settings-ui-plus-fuzzy-search';
+import { ObjectInput } from './settings-ui-plus-object-input';
+import { SettingsUIPlusArray } from './settings-ui-plus-array';
+import { SettingsUIPlusNavbar } from './settings-ui-plus-navbar';
+import { SettingsUIPreferenceContextMenuWidget } from './settings-ui-plus-preference-context-menu-widget';
+
+@injectable()
+export class SettingsUIPlusEditor extends ReactWidget {
+    static readonly ID = 'settings.ui.plus.editor';
+    static readonly LABEL = 'Settings UI+ Editor';
+    properties: any;
+    preferences: {
+        id: string;
+        name: string;
+        parent: any;
+        children: any[];
+    };
+
+    inputRefs: { [key: string]: React.RefObject<HTMLInputElement> } = {};
+    searchResults: any[];
+    defaultPreferences: any;
+    preferencesGroupNames = new Set<string>();
+    activeScope: number;
+    readonly scopes: PreferenceScope[] = PreferenceScope.getScopes().slice(1, 3);
+    @inject(PreferenceService) readonly service: PreferenceService;
+    @inject(PreferenceSchemaProvider) readonly schemaProvider: PreferenceSchemaProvider;
+    @inject(MessageService) readonly messageService: MessageService;
+
+    @inject(PreferenceConfigurations) protected readonly preferenceConfigs: PreferenceConfigurations;
+
+    @postConstruct()
+    protected init(): void {
+        this.id = SettingsUIPlusEditor.ID;
+        this.title.label = SettingsUIPlusEditor.LABEL;
+        this.defaultPreferences = this.schemaProvider.getCombinedSchema();
+        this.properties = this.defaultPreferences.properties;
+        // set active scope to [User] for now until we have a clearer idea of how to handle different settings
```

Comment:
What exactly would you like to clear out? Maybe someone here can help.

Reply:
If I remember correctly, it could've been about figuring out how to persist the current scope when the app is reset. My memory on this is hazy. @danieltomasku, do you have a better memory of this?

Reply:
@NicholasStenbeck I think this is actually a leftover comment from when we were hardcoding the preferences scope to User. Now that we've added the ability to toggle between User and Workspace scopes I believe this comment can be removed.